src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/EmailAddressAttribute.cs

```diff
@@ -30,20 +30,28 @@ public override bool IsValid(object value)
 
             // only return true if there is only 1 '@' character
             // and it is neither the first nor the last character
-            bool found = false;
-            for (int i = 0; i < valueAsString.Length; i++)
+            var index = valueAsString.IndexOf('@');
+           
+            if (index == 0) //cannot be the first char
             {
-                if (valueAsString[i] == '@')
-                {
-                    if (found || i == 0 || i == valueAsString.Length - 1)
-                    {
-                        return false;
-                    }
-                    found = true;
-                }
+                return false;
+            }
+            else if (index == -1) //char not found
+            {
+                return false;
+            }
+            else if (index != valueAsString.LastIndexOf('@')) //only one char
+            {
+                return false;
+            }
+            else if (index == valueAsString.Length - 1) //cannot be the last char
+            {
+                return false;
+            }
+            else
+            {
+                return true;
```

Comment:
Can't this just be:
```C#
int index = valueAsString.IndexOf('@');
return index > 0 && index != valueAsString.Length - 1 && index == valueAsString.LastIndexOf('@');
```
?

Reply:
I did something like this in the first place but thought a reader would prefer an stap by step check for readability. Also, I would put  `valueAsString.Length - 1` and `LasIndexOf` in the end as these require more computation. Up to you.

Reply:
I prefer the version I shared. Thanks.  If you want to split it across multiple lines, that's fine, e.g.
```C#
return
    index > 0 &&
    index != valueAsString.Length - 1 &&
    index == valueAsString.LastIndexOf('@');
```

Reply:
OK for
```
            return
                index > 0 &&
                index != valueAsString.Length - 1 &&
                index == valueAsString.LastIndexOf('@');
```